quantum/pointing_device/pointing_device_modes.c

```diff
@@ -0,0 +1,658 @@
+ *
+ * @return bool continue processing flag
+ */
+__attribute__((weak)) bool process_pointing_mode_user(pointing_mode_t pointing_mode, report_mouse_t* mouse_report) {
+    return true; // continue processing
+}
+
+/**
+ * @brief Keyboard level processing of pointing device modes
+ *
+ * Takes pointing_mode_t struct, and pointer to the report_mouse_t struct used in pointing mode process allowing
+ * modification of mouse_report directly and pointing mode using set_pointing_mode(pointing_mode) and the returned
+ * bool controls the continued processing of pointing modes.
+ *
+ * NOTE: returning false will stop mode processing (for overwriting modes)
+ *
+ * @params pointing_mode[in] pointing_mode_t
+ * @params mouse_report[in]  pointer to report_mouse_t
+ *
+ * @return bool continue processing flag
+ */
+__attribute__((weak)) bool process_pointing_mode_kb(pointing_mode_t pointing_mode, report_mouse_t* mouse_report) {
+    return true; // continue processing
+}
+
+/**
+ * @brief Weak function for user level adding of pointing mode divisors
+ *
+ * Takes in mode id and direction allowing for divisor values to be
+ * determined based on these parameters
+ *
+ * @params pointing_mode[in] uint8_t
+ * @params direction[in] uint8_t
+ *
+ * @return divisor uint8_t
+ */
+__attribute__((weak)) uint8_t get_pointing_mode_divisor_user(uint8_t mode_id, uint8_t direction) {
+    return 0; // continue processing
+}
+
+/**
+ * @brief Weak function for user level adding of pointing mode divisors
+ *
+ * Takes in mode id and direction allowing for divisor values to be
+ * determined based on these parameters
+ *
+ * @params pointing_mode[in] uint8_t
+ * @params direction[in] uint8_t
+ *
+ * @return divisor uint8_t
+ */
+__attribute__((weak)) uint8_t get_pointing_mode_divisor_kb(uint8_t mode_id, uint8_t direction) {
+    return 0; // continue processing
+}
+
+/**
+ * @brief Keyboard level modifying of divisors after being set and before use
+ *
+ * Allows modification the divisor after being set by get_pointing_mode_divisor stack before
+ * handing off to default post processing
+ *
+ * @params[in] pointer to divisor uint8_t
+ *
+ * @return bool continue processing flag
+ */
+__attribute__((weak)) bool pointing_mode_divisor_postprocess_kb(uint8_t* divisor) {
+    return true;
+}
+
+/**
+ * @brief User level modifying of divisors after being set and before use
+ *
+ * Allows modification the divisor after being set by get_pointing_mode_divisor stack before
+ * handing off to default post processing
+ *
+ * @params[in] pointer to divisor uint8_t
+ *
+ * @return bool continue processing flag
+ */
+__attribute__((weak)) bool pointing_mode_divisor_postprocess_user(uint8_t* divisor) {
+    return true;
+}
+
+#    if defined(SPLIT_POINTING_ENABLE) && defined(POINTING_DEVICE_COMBINED)
+/**
+ * @brief Check if left side is controlled by pointing mode framework
+ *
+ * @return is left side active [bool]
+ */
+bool is_pointing_mode_on_left(void) {
```

Comment:
hhmmm this function is primarily meant to access the local `is_left` variable and is not intended to be redefined as that local variable will be unavailable.

The intent is to for any extra user/keyboard added logic that changes which side is controlled by pointing modes using the `pointing_mode_switch_hands()` function and checking which is currently controlled by using `is_pointing_mode_on_left()`.

I see that the intent is does not communicate very well using a bool and toggle.
Perhaps I should set this up like this instead:
```c
// in pointing_device_modes.h
enum pointing_device_mode_side {PM_RIGHT_SIDE = 0, PM_LEFT_SIDE};

uint8_t get_pointing_mode_control_side(void);
void    set_pointing_mod_control_side(uint8_t side);

// code to handle this slight change added to pointing_device_modes.c and pointing_device.c
```
Would that be an improvement in readability and communicating the intent? (or is there a better way to handle this?)

Reply:
My intention was it'd allow you to easily enable a pointing mode on both sides simultaneously. Sure there are other ways to achieve this already. 

Anyway I missed this would need updating to allow for that.
```c
    if (is_pointing_mode_on_left()) {
        left_report = pointing_device_modes_task(left_report);
    } else {
        right_report = pointing_device_modes_task(right_report);
    }
```

Reply:
Latest update adds simultaneous independent pointing device control (still needs testing though).

Up to two pointing devices are supported natively if `SPLIT_POINTING_ENABLE` and `POINTING_DEVICE_COMBINED` are defined.

In fact an arbitrary number of pointing devices should be able to be supported but `pointing_device_task` code would need to be added to pass these on to the `pointing_device_mode_task` properly.